### ISSUE TITLE
idea: add linter to check that commited code is the same as that generated by go generate

### ISSUE THREAD

**Issue by mewmew** [golangci/golangci-lint#20]
For background, see the Reddit topic [Do you commit all the generated code in your Golang applications?](https://old.reddit.com/r/golang/comments/8mbi47/do_you_commit_all_the_generated_code_in_your/)

Specifically, @kemitche's reply https://old.reddit.com/r/golang/comments/8mbi47/do_you_commit_all_the_generated_code_in_your/dzmh9e3/

> We commit our generated code, but we also have our CI tools re-run code generation as a sanity check. If it finds any diffs, then we know the most recent commit failed to run code generation.
>
> We use dep to make sure that everyone is using the same version of code generation tools, and we try to stick to tools that are consistent when run repeatedly.

**Comment by golangci** [comment]
@mewmew thank you for suggestion!
I'm not sure that it should be placed in `golangci-lint`, it looks like 4 lines of shell code in CI:
```bash
#!/bin/bash
set -ue
go generate ./...
git diff --quiet || { echo "run go gonerate ./..."; exit 1 }
```

If I will include it into `golangci-lint`, what file and line to choose to report? For example, IDEs need file:line to correctly report an issue. The only option I see is next:
1. Check that `git diff` after `go generate ./...` reports changes.
2. Find all files with `//go:generate` comments.
3. Mark all lines in these files as issues: it can be performance-heavy. Also I'm not sure that we shouldn't also process dependencies of these files.
4. Pass through diff filter if it's enabled to leave only changed lines
5. Process all found issues by post-processor "max issues per linter" with limit 1: it will report only 1 file:line

**Comment by mewmew** [comment]
I think you're right, thanks for the suggestion. The shell script is simple enough not to warrant inclusion in golangci-lint, and as you mentioned it would be difficult to know what tools to invoke from the `//go:generate` comments.

I'll close this issue for now.

Cheers,
/u
